static/src/javascripts-legacy/projects/common/modules/check-dispatcher.js

```diff
@@ -43,34 +43,30 @@ define([
             return config.switches.emailInArticleOutbrain;
         },
         hasHighPriorityAdLoaded: function() {
-            return checkMediator.waitForCheck('thirdPartyTagsDisabled').then(function (thirdPartyTagsDisabled) {
-                // if thirdPartyTagsDisabled no external ads are loaded
-                if (thirdPartyTagsDisabled) {
-                    return false;
-                } else {
-                    return trackAdRender('dfp-ad--merchandising-high').then(function(highPriorityAdLoaded) {
-                        return highPriorityAdLoaded;
-                    });
-                }
-            });
+            // if thirdPartyTags false no external ads are loaded
+            if (commercialFeatures.thirdPartyTags) {
+                return trackAdRender('dfp-ad--merchandising-high').then(function(highPriorityAdLoaded) {
+                    return highPriorityAdLoaded;
+                });
+            } else {
+                return false;
```

Comment:
This is OK, the return value of this function (in this case - false) is used to resolve a corresponding promise that has been defined in check-mediator, the promise can be resolved with a boolean.